app/src/main/java/com/endava/flights/MainFlightActivity.kt

```diff
@@ -16,7 +28,13 @@ import androidx.fragment.app.FragmentActivity
  * Bonus track: Think one use case as a challenge for your partner.
  *
  */
-class MainFlightActivity : FragmentActivity() {
+class MainFlightActivity(private var viewModel : MainFlightViewModel = MainFlightViewModel()) : FragmentActivity() {
```

Comment:
The viewModel should be initialized outside activity constructor, the activity class it's an special one. We shouldn't touch the constructor.